app/repository/event.py

```diff
@@ -172,16 +172,15 @@ def create(db: Session, event: EventCreateDTO) -> str:
 
         family_import_id = new_family_event.family_import_id
 
-        # Generate the import_id for the new event
-        org = family_repo.get_organisation(db, cast(str, family_import_id))
-        if org is None:
-            raise ValidationError(
-                f"Cannot find counter to generate id for {family_import_id}"
-            )
-
-        org_name = cast(str, org.name)
-
         if not new_family_event.import_id:
+            # Generate the import_id for the new event
+            org = family_repo.get_organisation(db, cast(str, family_import_id))
```

Comment:
This block of code could probably go into it's own function - seems the document block above is essentially the same behaviour? Should be able to pass in the bits you need to into a helper function. DRY :)

Reply:
I'll do this as a separate PR as it is not part of this work.